bundles/SimpleBackendSearchBundle/src/Migrations/Version20230914085710.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\SimpleBackendSearchBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\Migrations\AbstractMigration;
+
+final class Version20230914085710 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return 'Change some columns of search_backend_table to non nullable';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $this->addSql('ALTER TABLE `search_backend_data` MODIFY `fullpath` varchar(765) CHARACTER SET ascii COLLATE ascii_general_ci NOT NULL');
+        $this->addSql('ALTER TABLE `search_backend_data` MODIFY `type` varchar(20) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NOT NULL');
+        $this->addSql('ALTER TABLE `search_backend_data` MODIFY `subtype` varchar(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_general_ci NOT NULL');
```

Comment:
- [ ] before turning to non-nullable, needs to set all the former `null` to empty string eg.
`UPDATE `search_backend_data` SET fullpath = '' WHERE fullpath IS NULL;`
or would trigger `Data truncated` error
![image](https://github.com/pimcore/pimcore/assets/6014195/54f5226d-f813-4d47-a6fa-0f40ffee5826)

- [ ] the collation seeems wrong, should be `utf8_general_ci`
https://github.com/pimcore/pimcore/blob/855cefd1a1ed67f8fa399eb042a5c9c7d8e72f2c/bundles/InstallBundle/dump/install.sql#L343C53-L343C53
- [ ] `install.sql` should be also adapted to use the right table column definitions on fresh installation https://github.com/pimcore/pimcore/blob/855cefd1a1ed67f8fa399eb042a5c9c7d8e72f2c/bundles/InstallBundle/dump/install.sql#L343C1-L346

Reply:
Don't know how I ended up in our active Pimcore installation with `ascii` charset and collation. It's an old installation (6 years+), so probably missed some migration before.
I have setup an empty skeleton installation and can confirm the collation is set to `utf8_general_ci` when inspecting the DB schema.
Changed the migration now to match this.